capstone/src/main/java/com/google/step/servlets/LoginServlet.java

```diff
@@ -0,0 +1,113 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that will handle login information.
+ */
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+
+    UserService userService = UserServiceFactory.getUserService();
+    Map<String, Object> login = new HashMap<>();
+
+    if (userService.isUserLoggedIn()) {
+      String useremail = userService.getCurrentUser().getEmail();
+      String type = checkIfRestaurantOrUser(useremail);
+
+      // If user is loggedin but not in our system.
+      if (type.equals("None")) {
+        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+        return;
+      }
+
+      String urltoRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urltoRedirectToAfterUserLogsOut);
+
+      login.put("Loggedin", true);
+      login.put("Type", type);
+      login.put("Email", useremail);
+      login.put("logOutUrl", logoutUrl);
+    } else {
+      String urltoRedirectToAfterUserLogIn = "/";
+      String urltoRedirectAfterUserSignUp = "/user-signup.html";
+      String urltoRedirectAfterRestaurantSignUp = "/restaurant-signup.html";
+
+      String loginUrl = userService.createLoginURL(urltoRedirectToAfterUserLogIn);
+      String userSignUpUrl = userService.createLoginURL(urltoRedirectAfterUserSignUp);
+      String restaurantSignUpUrl = userService.createLoginURL(urltoRedirectAfterRestaurantSignUp);
+
+      login.put("Loggedin", false);
+      login.put("loginUrl", loginUrl);
+      login.put("userSignUpUrl", userSignUpUrl);
+      login.put("restaurantSignUpUrl", restaurantSignUpUrl);
+    }
+
+    String json = convertToJsonUsingGson(login);
+    response.getWriter().println(json);
+  }
+
+  // TODO: Change to have object return and return User or Restaurant Object
+  /**
+   * Checks if the current email is from a user or a restaurant.
+   * @param email of the current user.
+   * @return String containing the words User or Restaurant.
+   */
+  private static String checkIfRestaurantOrUser(String email) {
+    boolean restaurant = checkUserTypeExists(email, "Restaurant");
+    boolean user = checkUserTypeExists(email, "User");
+    if (restaurant == true && user == false) {
+      return "Restaurant";
+    } else if (restaurant == false && user == true) {
+      return "User";
+    }
+    return "None";
```

Comment:
Is it ever possible for it to be anything other than one or the other? If not, then don't bother with "none" - just throw an IllegalStateException.

If you enforce uniqueness and correctness with transactions right from the start, then you'll avoid all the hackery of dealing with an "if(none)" case all over the place.

Reply:
Deleted the method due to the change in user-restaurant dynamic.